ui/pages/confirmations/components/confirm/info/shared/redesigned-gas-fees/redesigned-gas-fees.tsx

```diff
@@ -0,0 +1,104 @@
+import { useSelector } from 'react-redux';
+import { EditGasModes } from '../../../../../../../../shared/constants/gas';
+import {
+  addHexes,
+  multiplyHexes,
+} from '../../../../../../../../shared/modules/conversion.utils';
+import { ConfirmInfoSection } from '../../../../../../../components/app/confirm/info/row/section';
+import { currentConfirmationSelector } from '../../../../../../../selectors';
+import EditGasPopover from '../../../../edit-gas-popover';
+import { useEIP1559TxFees } from '../../hooks/useEIP1559TxFees';
+import { useSupportsEIP1559 } from '../../hooks/useSupportsEIP1559';
+import { FeesDetails } from './fees-details';
+
+function getGasEstimate(
+  transactionMeta: TransactionMeta,
+  supportsEIP1559: boolean,
+): string {
+  let { gas: gasLimit, gasPrice } = transactionMeta.txParams;
+  const { estimatedBaseFee } = transactionMeta.txParams;
+
+  // override with values from `dappSuggestedGasFees` if they exist
+  gasLimit = transactionMeta.dappSuggestedGasFees?.gas || gasLimit || '0x0';
+  gasPrice =
+    transactionMeta.dappSuggestedGasFees?.gasPrice || gasPrice || '0x0';
+  const maxPriorityFeePerGas =
+    transactionMeta.dappSuggestedGasFees?.maxPriorityFeePerGas ||
+    transactionMeta.txParams?.maxPriorityFeePerGas ||
+    '0x0';
+
+  let gasEstimate;
+  if (supportsEIP1559) {
+    // Minimum Total Fee = (estimatedBaseFee + maxPriorityFeePerGas) * gasLimit
+    const minimumFeePerGas = addHexes(
+      estimatedBaseFee || '0x0',
+      maxPriorityFeePerGas,
+    );
+
+    gasEstimate = multiplyHexes(minimumFeePerGas, gasLimit);
+  } else {
+    gasEstimate = multiplyHexes(gasLimit, gasPrice);
+  }
+
+  return gasEstimate;
+}
+
+const Type0TxGasModal = ({
+  closeCustomizeGasPopover,
+  transactionMeta,
+}: {
+  closeCustomizeGasPopover: () => void;
+  transactionMeta: TransactionMeta;
+}) => {
+  return (
+    <EditGasPopover
+      onClose={closeCustomizeGasPopover}
+      mode={EditGasModes.modifyInPlace}
+      transaction={transactionMeta}
+    />
+  );
+};
+
+export const RedesignedGasFees = ({
+  showAdvancedDetails,
+}: {
+  showAdvancedDetails: boolean;
+}) => {
+  const transactionMeta = useSelector(
+    currentConfirmationSelector,
+  ) as TransactionMeta;
+
+  const [showCustomizeGasPopover, setShowCustomizeGasPopover] = useState(false);
+  const closeCustomizeGasPopover = () => setShowCustomizeGasPopover(false);
+
+  const { supportsEIP1559 } = useSupportsEIP1559(transactionMeta);
+
+  const { maxFeePerGas, maxPriorityFeePerGas } =
+    useEIP1559TxFees(transactionMeta);
+
+  const gasEstimate = getGasEstimate(transactionMeta, supportsEIP1559);
+
+  if (!transactionMeta?.txParams) {
+    return null;
+  }
+
+  return (
+    <ConfirmInfoSection>
+      <FeesDetails
+        gasEstimate={gasEstimate}
+        maxFeePerGas={maxFeePerGas}
+        maxPriorityFeePerGas={maxPriorityFeePerGas}
+        setShowCustomizeGasPopover={setShowCustomizeGasPopover}
+        showAdvancedDetails={showAdvancedDetails}
```

Comment:
The more instances of this we get, the more beneficial that `useIsAdvancedDetailsEnabled` context would have.